### DISCUSSION TITLE
Migrate from a external library to internal without re-uploading all the images

### DISCUSSION THREAD

**Discussion by TheZoker** [immich-app/immich#14378] [Q&A]
Hi there,

I'm currently using Synology Photos to backup photos to my NAS. 
I have a folder structure that looks like this:
- {year}/{album} (or if no album just a "Others") folder
- A {year}/upload folder for newly uploaded photos, which are then manually moved to the album or the others folder
- No photos are in the {year} folder directly 

The album folders are manually created folders, where I move fitting images.
I change the upload folder every year to fit the {year}/upload folder.

I also run immich on a lxc container within proxmox with the nas folder mounted via NFS like this:
```
/shared/nas/photos:/mnt/media/nas
```

Within immich I added this folder as a external library. 

Since I really like immich and I want to remove synology photos completely and move everything to immich, my question would be what is the best way forward. 
I added a storage path like this:
```
{{y}}/{{#if album}}{{album}}{{else}}Other{{/if}}/{{filename}}
```

How can I switch from a external library to a internal library without having to re-upload all the images again? 

In my docker compose file I guess a general volume mapping like this:
```
      - ~/files/immich/backups:/usr/src/app/upload/backups
      - ~/files/immich/encoded-video:/usr/src/app/upload/encoded-video
      - ~/files/immich/profile:/usr/src/app/upload/profile      
      - ~/files/immich/thumbs:/usr/src/app/upload/thumbs
      - ~/files/immich/upload:/usr/src/app/upload/upload
      - /shared/nas/photos:/usr/src/app/upload/library/admin
```

I do not want to have all the metadata (thumbs, profile images etc.) on the NAS, only the real photos. 

So what is the best way to migrate from a external library to a internal one?
Thanks!

**Comment by bo0tzz** [comment]
There's no way to do this at the moment.

**Reply by TheZoker** [reply]
Is this maybe planned?

**Comment by mmomjian** [comment]
The only way to use internal is to re upload everything using the immich CLI, immich go, or web browser.

**Reply by TheZoker** [reply]
I see thanks for the hint!

**Comment by petersuba** [comment]
What would be the additional benefit of doing that? Curious as I currently also map a large image library on Synology as external on immich - and it seems to work fine.

**Reply by TheZoker** [reply]
I want to only use immich and if I use the external library, I cannot upload any images to it. 
And I do not want to have two different locations, where I store my images (an internal and an external one)

**Reply by petersuba** [reply]
>  if I use the external library, I cannot upload any images to it. 
I am not sure what you mean by this. I constantly upload photos to my external library an immich picks them up on the nightly scan. Of course, it isn't instantenous, that's true...

Not trying to dissuade you from moving fully to immich, just truly trying to see if there is a benefit I can't see yet.

**Reply by TheZoker** [reply]
How do you upload images to the external library? With immich or with another software?

**Reply by petersuba** [reply]
Depends on the source of the image. There are historical libraries or photos from photoshoots I am simply uploading to the folder structure via Synology file manager, and from mobile I am uploading via Synology photos doing the backup. I may be migrating away from that at some point so I have a single app to rely on once I know I can utilise/configure immich mobile app properly in my scenario, but that's not the case yet.

**Comment by eharris** [comment]
This appears to be a duplicate thread of #12562
